blocksync/blocksync.go

```diff
@@ -8,40 +8,39 @@ package blocksync
 
 import (
 	"context"
-	"net"
 
 	"github.com/golang/protobuf/proto"
+	peerstore "github.com/libp2p/go-libp2p-peerstore"
 	"go.uber.org/zap"
 
 	"github.com/iotexproject/iotex-core/actpool"
 	"github.com/iotexproject/iotex-core/blockchain"
 	"github.com/iotexproject/iotex-core/blockchain/block"
 	"github.com/iotexproject/iotex-core/config"
-	"github.com/iotexproject/iotex-core/p2p/node"
 	"github.com/iotexproject/iotex-core/pkg/lifecycle"
 	"github.com/iotexproject/iotex-core/pkg/log"
 	"github.com/iotexproject/iotex-core/pkg/routine"
 	"github.com/iotexproject/iotex-core/proto"
 )
 
 type (
-	// Unicast sends a unicast message to the given address
-	Unicast func(addr net.Addr, msg proto.Message) error
+	// UnicastOutbound sends a unicast message to the given address
+	UnicastOutbound func(ctx context.Context, peer peerstore.PeerInfo, msg proto.Message) error
```

Comment:
I wonder if it's necessary to let app pass in the context. What app level context should be sent? If no, is it enough for p2p agent to compose a context with network info there?